counter-reset: new example and description (#30440)
mdn/content: files/en-us/web/css/counter-reset/index.md

After creating a counter using `counter-reset`, you can adjust its value by using the {{cssxref("counter-set")}} property. This is counterintuitive because, despite its name, the `counter-reset` property is used for creating and initializing counters, while the `counter-set` property is used for resetting the value of an existing counter.

Setting `counter-increment: none` on a selector with greater specificity overrides the creation of the named counter set on selectors with lower specificity.

### Default initial values

The default initial values of both regular and reversed counters make it easy to implement the two most common numbering patterns: counting up from one to the number of elements and counting down from the number of elements to one, respectively. By including a counter value for a named counter, your counter can count up or down, starting at an integer value.

Regular counters default to `0` if no reset value is provided. By default, regular counters increment by one, which can be adjusted with the {{cssxref("counter-increment")}} property.

```css
h1 {

>   /* Creates and sets the chapter and page counters to 0,
>      and the section counter to 4 */

Reviewer: ```suggestion
  /* Create the counters "chapter" and "page" and set to initial default value.
     Create the counter "section" and set to "4". */
```

  counter-reset: chapter section 4 page;
}
```

### Reversed counters

When creating reversed counters without a value, the counter will start with the value equal to the number of elements in the set, counting down so the last element in the set is `1`. By default, reverse counters decrement by one; this can also be changed with the `counter-increment` property.

```css
h1 {

>   /* Create reversed counters chapter and section, setting
>      the chapter counter as the number of elements and the section counter as 10.
>      Create the pages counter with initial default value. */

Reviewer: ```suggestion
  /* Create reversed counters "chapter" and "section".
      Set "chapter" as the number of elements and "section" as "10".
      Create the counter "pages" with the initial default value. */
```

  counter-reset: reversed(chapter) reversed(section) 10 pages;
}
```

> ### Built-in list-item counter

Reviewer: ```suggestion
### Built-in `list-item` counter
```

  - : No counter reset is to be performed.
    This can be used to override a `counter-reset` defined in a less specific rule.

### Description

> 

Reviewer: - This para from the previous version was useful. Could we add it to the Description? (changed "normal" to "regular")

> "Regular counters have a default initial value of `0`. Reversed counters have a default initial value set to the number of elements at the current level. "

- For better content flow, can we organize the "Description" into subsections like:
   - Using the default initial values
   - Using `counter-increment` to modify default settings
   - Adjusting the built-in `list-item` counter

## Examples

### Overriding the `list-item` counter

In this example, the `counter-reset` property is used to set a starting value for an implicit `list-item` counter.

#### HTML

```html
<ol>
  <li>First</li>
  <li>Second</li>
  <li>Third</li>
  <li>Fourth</li>
  <li>Fifth</li>
</ol>
```

#### CSS

Using `counter-reset`, we set the implicit `list-item` counter to start at a value other than the default `1`:

```css
ol {
  counter-reset: list-item 3;
}
```

#### Result

> {{EmbedLiveSample("Overriding implicit list item counters", 140, 300)}}

Reviewer: ```suggestion
{{EmbedLiveSample("Overriding the list-item counter", 140, 300)}}
```
